webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go

```diff
@@ -462,6 +464,58 @@ func (v validator) validateNetworkInterfaceSpec(
 	return allErrs
 }
 
+// mtu and routes is available only with CloudInit bootstrap providers.
+// nameservers and searchDomains is available only with the following bootstrap
+// providers: CloudInit, LinuxPrep, and Sysprep (except for RawSysprep).
+func (v validator) validateNetworkSpecWithBootstrap(
+	interfacePath *field.Path,
+	interfaceSpec vmopv1.VirtualMachineNetworkInterfaceSpec,
+	vm *vmopv1.VirtualMachine) field.ErrorList {
+
+	var allErrs field.ErrorList
+	cloudInit := vm.Spec.Bootstrap.CloudInit
+	linuxPrep := vm.Spec.Bootstrap.LinuxPrep
+	sysPrep := vm.Spec.Bootstrap.Sysprep
+
+	if mtu := interfaceSpec.MTU; mtu != nil && cloudInit == nil {
+		allErrs = append(allErrs, field.Invalid(
+			interfacePath.Child("mtu"),
+			"mtu",
```

Comment:
I think typically elsewhere for `field.Invalid`'s we put the value here instead.